src/lints/struct_marked_non_exhaustive_changed_type.ron

```diff
@@ -0,0 +1,74 @@
+SemverQuery(
+    id: "struct_marked_non_exhaustive_changed_type",
+    human_readable_name: "struct marked #[non_exhaustive] with no fields of any kind became an enum or union",
+    description: "A struct marked #[non_exhaustive] with no fields of any kind was converted into an enum or union, breaking pattern matching.",
```

Comment:
nitpick: "of any kind" will probably be confusing to our users. Enum variants have kinds: plain, tuple, or struct variant. Struct fields don't really have a "kind" — they are merely public or not.

Let's avoid mentions of "kind" throughout.
```suggestion
    human_readable_name: "struct marked #[non_exhaustive] with no fields became an enum or union",
    description: "A struct marked #[non_exhaustive] with no fields was converted into an enum or union, breaking pattern matching.",
```